### DISCUSSION TITLE
privateGPT killed while launching in Ubuntu 22.10. No error message

### DISCUSSION THREAD

**Discussion by cslibby** [imartinez/privateGPT#477] [General]
I have been required to use a virtual environment. Here is what I get

(privateGPT) cslibby@Linux-Thinkpad:~/privateGPT$ ./bin/python privateGPT.py
Using embedded DuckDB with persistence: data will be stored in: db
Found model file.
gptj_model_load: loading model from 'models/ggml-gpt4all-j-v1.3-groovy.bin' - please wait ...
gptj_model_load: n_vocab = 50400
gptj_model_load: n_ctx   = 2048
gptj_model_load: n_embd  = 4096
gptj_model_load: n_head  = 16
gptj_model_load: n_layer = 28
gptj_model_load: n_rot   = 64
gptj_model_load: f16     = 2
gptj_model_load: ggml ctx size = 5401.45 MB
gptj_model_load: kv self size  =  896.00 MB
gptj_model_load: .................................Killed

**Comment by cslibby** [comment]
I have been trying to determine if this is a memory problem or the virtual environment in my Ubuntu. I was able to get it to work on my Windows machine, and I really want it to be available on my Linux Machine. I am trying to use this to help with analyzing my data for my dissertation.

**Comment by cslibby** [comment]
Ubuntu 22.10 on a Lenovo Laptop T490. Any help would be appreciated.

**Comment by bdbais** [comment]
I have same error on a VPS with 4GB RAM:

privateGPT# python3.10 privateGPT.py
Found model file at  models/ggml-gpt4all-j-v1.3-groovy.bin
gptj_model_load: loading model from 'models/ggml-gpt4all-j-v1.3-groovy.bin' - please wait ...
gptj_model_load: n_vocab = 50400
gptj_model_load: n_ctx   = 2048
gptj_model_load: n_embd  = 4096
gptj_model_load: n_head  = 16
gptj_model_load: n_layer = 28
gptj_model_load: n_rot   = 64
gptj_model_load: f16     = 2
gptj_model_load: ggml ctx size = 5401.45 MB
gptj_model_load: kv self size  =  896.00 MB
gptj_model_load: ............................Killed

I've used Docker solution to use it, on a bid workstation :) How much memory/cpu is required for to start a little demo?

**Reply by thanhtantran** [reply]
I think it required 16GB RAM

**Comment by abundis-rmn2** [comment]
Its there a smaller model, that can run on 16gb machines?

**Reply by abundis-rmn2** [reply]
FYI, i made a swap file, and now is working, my pc only have 8gb of ram and i added a 4gb swapfile
Model is slow as hell but it works.

--
https://www.cloudsigma.com/adding-swap-space-on-ubuntu-20-04-a-tutorial/

**Comment by PasunuriSrinidhi** [comment]
@cslibby  can you please give the steps to get work on ubuntu, as i am getting this error                                                             
    Defaulting to user installation because normal site packages is not writeable
ERROR: Could not find a version that satisfies the requirement gpt4all==1.0.8 (from versions: 0.1.5, 0.1.6, 0.1.7).         
Thank you !
